scrapper.py

```diff
@@ -31,67 +42,176 @@ class Crawler:
     """
     Crawler implementation
     """
-    def __init__(self, seed_urls: list, max_articles: int):
-        pass
+
+    def __init__(self, seed_urls: list, max_articles: int, max_articles_per_seed: int):
+        self.seed_urls = seed_urls
+        self.max_articles = max_articles
+        self.max_articles_per_seed = max_articles_per_seed
+        self.urls = []
 
     @staticmethod
     def _extract_url(article_bs):
-        pass
+        return article_bs.find('a').attrs['href']
 
     def find_articles(self):
         """
         Finds articles
         """
-        pass
+        self.get_search_urls()
+        for seed_url in urls:
+            try:
+                response = requests.get(seed_url, headers=headers)
+                sleep(random.randint(2, 6))
+                response.encoding = 'utf-8'
+                if not response:
+                    raise IncorrectURLError
+
+            except IncorrectURLError:
+                continue
+
+            page_content = response.content
+            page_bs = BeautifulSoup(page_content, features='lxml')
+            page_urls = page_bs.find_all('div', {'class': 'news-preview-content'})
+
+            urls_number = min(max_articles_num_per_seed, len(page_urls), (max_articles_num - len(self.urls)))
+            for index in range(urls_number):
+                self.urls.append('https://vn.ru' + self._extract_url(article_bs=page_urls[index]))
 
     def get_search_urls(self):
         """
         Returns seed_urls param
         """
-        pass
+
+        themes = ['ekonomika', 'oblast', 'obshchestvo',  'proisshestviya', 'dom', 'transport', 'razvlecheniya']
+        self.seed_urls.extend([f'{self.seed_urls[0]}news/{theme}/' for theme in themes])
 
 
 class ArticleParser:
     """
     ArticleParser implementation
     """
+
     def __init__(self, full_url: str, article_id: int):
-        pass
+        self.full_url = full_url
+        self.article_id = article_id
+        self.article = Article(full_url, article_id)
 
     def _fill_article_with_text(self, article_soup):
-        pass
+
+        article_text = article_soup.find('div', {'class': 'js-mediator-article divider-news clearfix'}).find_all('p')
+
+        if not re.sub(r'[^А-Яа-я]', '', ''.join(list(map(lambda x: x.text, article_text)))):
```

Comment:
why do we need this condition?

Reply:
Well, I can just leave it but, in this case, titles from other news appear in the text.
![2021-03-12_16-47-24](https://user-images.githubusercontent.com/37346407/110948859-ec4da300-8352-11eb-8834-708c5bc294e8.png)